addon/components/task-button.js

```diff
@@ -0,0 +1,62 @@
+import { inject as service } from "@ember/service";
+import Component from "@glimmer/component";
+import { queryManager } from "ember-apollo-client";
+import { lastValue, dropTask } from "ember-concurrency-decorators";
+
+import allWorkItems from "ember-caluma/gql/queries/all-work-items.graphql";
+
+/**
+ * Component to render TaskButton which mutates the work item through the task slug and filters.
+ *
+ * ```hbs
+ * <TaskButton @mutation="complete" @task="some-task" filters=(Optional)/>
+ * ```
+ *
+ * @class TaskButtonComponent
+ */
+export default class TaskButtonComponent extends Component {
+  /**
+   * The mutation which is going to be used for the work item.
+   * Valid mutations are: complete, skip, cancel
+   * @argument {String} mutation
+   */
+  /**
+   * The task slug of the work item which should be mutated.
+   * @argument {String} task
+   */
+  /**
+   * The filters to find the work item which should be mutated.
+   * @argument {Array} filters
+   */
+
+  @queryManager apollo;
+
+  @service notification;
+  @service intl;
+
+  @lastValue("fetchWorkItem") workItem;
+
+  @dropTask
+  *fetchWorkItem() {
+    try {
+      const res = yield this.apollo.query(
+        {
+          query: allWorkItems,
+          variables: {
+            filter: [
+              { taskSlug: this.args.task },
+              ...(this.args.filters || []),
+            ],
+          },
+        },
+        "allWorkItems.edges"
```

Comment:
```suggestion
        "allWorkItems.edges.firstObject.node"
```

Then directly return it

Reply:
Sadly doesn't work, firstObject doesn't exists at that point

Reply:
Hm okay.. Then use optional chaining below: `return res[0]?.node;` otherwise we show an error if no work item exists